From fcce1284c6c4a8745ad1f2e02899f7b498fd5be1 Mon Sep 17 00:00:00 2001
From: Takuro <ssk.ktai@gmail.com>
Date: Sun, 27 Sep 2026 16:07:53 +0900
Subject: [PATCH 1/2] PhoneSwitcher: Skip setPreferredDataModem on single SIM
 devices

Calling setPreferredDataModem via IRadioConfig on single SIM devices (mActiveModemCount <= 1) causes issues on legacy modem HAL implementations that do not support multi-SIM data switching commands. Align this behavior with the existing mActiveModemCount > 1 check used for setDataAllowed.

Change-Id: Ia6582eb8d797fad85d69e7d65bdde9ca753b50a7
---
 .../com/android/internal/telephony/data/PhoneSwitcher.java  | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/java/com/android/internal/telephony/data/PhoneSwitcher.java b/src/java/com/android/internal/telephony/data/PhoneSwitcher.java
index 4d4f0a91fe..2ff167adf3 100644
--- a/src/java/com/android/internal/telephony/data/PhoneSwitcher.java
+++ b/src/java/com/android/internal/telephony/data/PhoneSwitcher.java
@@ -1422,8 +1422,10 @@ public class PhoneSwitcher extends Handler {
             }
         } else if (phoneId == mPreferredDataPhoneId) {
             // Only setPreferredDataModem if the phoneId equals to current mPreferredDataPhoneId
-            logl("sendRilCommands: setPreferredDataModem - phoneId: " + phoneId);
-            mRadioConfig.setPreferredDataModem(mPreferredDataPhoneId, message);
+            if (mActiveModemCount > 1) {
+                logl("sendRilCommands: setPreferredDataModem - phoneId: " + phoneId);
+                mRadioConfig.setPreferredDataModem(mPreferredDataPhoneId, message);
+            }
         }
     }
 
-- 
2.55.0

